plugins/jobtype/src/azkaban/jobtype/tuning/TuningParameterUtils.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright 2012 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.jobtype.tuning;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.httpclient.HttpStatus;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.apache.log4j.Logger;
+
+import azkaban.jobtype.HadoopConfigurationInjector;
+import azkaban.utils.Props;
+
+import com.google.gson.Gson;
+
+/**
+ * TuningParameterUtils is a utility class responsible for updating tuned hadoop parameters for the job
+ * by calling tuning api. Before calling tuning api, it gets default parameters for the job and passes default
+ * parameters to api. By default it tries 3 times in case api is down before running the job with default parameters.
+
+ */
+public class TuningParameterUtils {
+
+  private static Logger log = Logger.getRootLogger();
+  private static final int DEFAULT_TUNING_API_RETRY_COUNT = 3;
+  private static final int DEFAULT_TUNING_API_TIMEOUT_MS = 10000;
+
+  private static final String USER_TO_PROXY = "user.to.proxy";
```

Comment:
There are constants here that it seems must be defined elsewhere?

Reply:
These are specifics to tuning API and will not be used anywhere else, so defined here to avoid confusion

Reply:
`USER_TO_PROXY` is not tuning API-specific, is it?

Reply:
Finally found HadoopSecurityManager.USER_TO_PROXY. Using this now. 
Passing as a user to tuning API.